oxide/lib.go

```diff
@@ -22,71 +21,86 @@ const TokenEnvVar = "OXIDE_TOKEN"
 // HostEnvVar is the environment variable that contains the host.
 const HostEnvVar = "OXIDE_HOST"
 
+// defaultUserAgent is the user agent set for the client.
+// TODO: Dynamically source this from ../VERSION. Embedding doesn't work since
+// it uses `..` in the path. Perhaps a compile-time variable?
+const defaultUserAgent = "Oxide Go SDK v0.1.0-beta3"
+
+// Config is the configuration that can be set on a Client.
+type Config struct {
+	// Base URL of the Oxide API including the scheme. For example,
+	// https://api.oxide.computer.
+	Address string
+
+	// Oxide API authentication token.
+	Token string
+
+	// A custom HTTP client to use for the Client instead of the default.
+	HTTPClient *http.Client
+
+	// A custom user agent string to add to every request instead of the
+	// default.
+	UserAgent string
```

Comment:
This is really nice! I like the concise documentation

Reply:
Thank you! I appreciate the kind words!